rundeckapp/src/main/groovy/org/rundeck/app/authorization/BaseAuthContextProvider.groovy

```diff
@@ -91,4 +93,21 @@ class BaseAuthContextProvider implements AuthContextProvider {
             authorizationService.getAuthorizationForSubject(new SubjectUserAndRoles(subject))
         )
     }
+
+    @Override
+    UserAndRolesAuthContext getAuthContextForUrnProject(String project) {
+        if (!project) {
+            throw new RuntimeException(
+                    "getAuthContextForUrnProject: Cannot get AuthContext without project: " +
+                            "${project}"
+            )
+        }
+
+        Subject subject = AclsUtil.getSubjectUrn(project)
+
+        return new SubjectAuthContext(
```

Comment:
"UserAndRoles" and "UserAndRolesAuthContext" interfaces are used in some places that expect a username and roles, please review locations where it is used.  Perhaps a getUrn() can be added to the "UserAndRoles" interface so that the urn can be exposed.  For example, see AuthContextEvaluatorCacheManager.groovy